packages/now-cli/test/dev/integration.js

```diff
@@ -265,6 +265,36 @@ test(
   })
 );
 
+test(
+  '[now dev] test cleanUrls serve correct content',
+  testFixtureStdio('test-clean-urls', async (t, port) => {
+    const opts = { redirect: 'manual' };
+    const testPath = async (status, path, expectedText, headers = {}) => {
+      const res = await fetch(`http://localhost:${port}${path}`, opts);
+      t.is(res.status, status);
+      if (expectedText) {
+        const actualText = await res.text();
+        t.is(actualText.trim(), expectedText.trim());
+      }
+      if (headers) {
+        Object.keys(headers).forEach(key => {
+          const k = key.toLowerCase();
+          t.is(headers[k], res.headers[k]);
+        });
+      }
+    };
+    await testPath(200, '/', 'Index Page');
+    await testPath(200, '/about', 'About Page');
+    await testPath(200, '/sub', 'Sub Index Page');
+    await testPath(200, '/sub/another', 'Sub Another Page');
+    await testPath(200, '/style.css', 'body { color: green }');
+    await testPath(301, '/index.html', '', { Location: '/' });
+    await testPath(301, '/about.html', '', { Location: '/about' });
+    await testPath(301, '/sub/index.html', '', { Location: '/sub' });
```

Comment:
This one should be `Location: /sub/`. Consider relative file paths to assets in the HTML file (images, scripts, etc.). If it's this way in production then I argue that it's incorrect and we need to fix it.

Reply:
@TooTallNate We have a separate toggle for that called `trailingSlash: true`.

Indeed, this is how it works in production right now.

Reply:
Developers have to choose to either add or remove a trailing slash consistently. What you're asking for would have a trailing slash for index but not otherwise. This would mean you couldn't rename `about.html` to `about/index.html` without first changing the URL. This allows relative URLs to work consistently (maybe not as a user would initially expect but this can be documented)

Reply:
Why would `<img src="foo.png" />` for `docs/about.html` point to `docs/foo.png`, but in `docs/index.html` would point to top-level `foo.png` (since it's being accessed at `/docs` without a trailing slash, it _appears_ to be in the top-level and the browser resolves assets relative to the top-level)?

Reply:
Because relative URLs are based on the page's URL not where it's located in the filesystem

Reply:
So the web developer needs to have a different mental model when writing the `about.html` page vs. the `index.html` page, as relative assets will be resolved to different directories. Seems very confusing to me.

Reply:
Previously, a user could add or remove a slash to the URL and get 404s

Reply:
Sure, but relative URLs still do exist. The difference in mental model just because a file is named `index.html` is very unfortunate to me.

Reply:
I think we can guide the user to use absolute urls instead of relative. Something like `<img src="/images/foo.png" />` would work for any file in any directory. The idea of `cleanUrls` and `trailingSlash` is basically intended to change the relative path because why else would the user enable it? As long as we ensure `now dev` and `now` production deployments have the same behavior, there won't be any unexpected results.